### PR TITLE
adding resource to generate an access token

### PR DESCRIPTION
**Pull request by nmonterroso**
@Wikia/services-team 
https://wikia-inc.atlassian.net/browse/SERVICES-893

Adding `POST /users/{userId}/tokens` resource, ~~which needs to be authenticated with `THE-SCHWARTZ`~~, and will generate an access token for the specified `userId`

### CODE REVIEW

File: controllers/users_controller.go [hunk 1]
```diff
@@ -327,3 +330,12 @@ func (controller *UsersController) forceLogoutAllSessions(w http.ResponseWriter,
 		utils.NoContent().Write(w)
 	}
 }
+
+func (controller *UsersController) forcedTokenGenerationHandler(w http.ResponseWriter, r *http.Request, params map[string]string) {
```
Comment (ArturKlajnerok) [inline controllers/users_controller.go]: Can we create new `token_controller.go` and start to move there resources like this one, so we can deprecate other controllers at some point of breaking it apart ?

File: controllers/users_controller.go [hunk 2]
```diff
@@ -327,3 +330,12 @@ func (controller *UsersController) forceLogoutAllSessions(w http.ResponseWriter,
...
+	resp := controller.server.NewResponse()
+	defer resp.Close()
+	defer osin.OutputJSON(resp, w, r)
+	userId := utils.ParseInt64(params["user_id"])
+
+	addAccessTokenToResponse(controller, userId, models.AuthenticatedNormalPass, resp, r)
```
Comment (ArturKlajnerok) [inline controllers/users_controller.go]: Can we create json with access token directly in redis for this resource and move away from using `osin.OutputJSON` and `FinishAccessRequest`? If we add  valid `AccessData` via `SaveAccess()` the token should valid for further usage.

Comment (nmonterroso) [inline controllers/users_controller.go]: `AccessData` is also an `osin` package struct, so we aren't really reducing our reliability on `osin` if we were to do that. Also in doing that we'd have to call `GenerateAccessToken` in the `osin` package. 

I agree we should reduce our dependency on `osin`, but we should do that as a separate effort where we have a clear plan

File: controllers/users_controller.go (hunk 1 above)
Comment (nmonterroso) [inline controllers/users_controller.go]: good idea. added in 0eba63b

File: controllers/token_controller_e2e_test.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
+package controllers
+import (
+	"github.com/stretchr/testify/suite"
+	"github.com/Wikia/go-commons/apiclient"
+	"github.com/Wikia/helios/helpers/testhelpers"
+	"github.com/Wikia/helios/helpers"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"github.com/Wikia/helios/models"
+"testing"
```
Comment (sqreek) [inline controllers/token_controller_e2e_test.go]: formatter

Comment (nmonterroso) [inline controllers/token_controller_e2e_test.go]: 76d5ee8

File: controllers/users_controller.go (hunk 1 above)
Comment (sqreek) [inline controllers/users_controller.go]: Is this should be removed?
it's duplicated code

Comment (nmonterroso) [inline controllers/users_controller.go]: Oops, I forgot to remove it.

Comment (nmonterroso) [inline controllers/users_controller.go]: 3268375

File: controllers/token_controller.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+func (c *TokenController) GetServer() *osin.Server {
+	return c.osin
+}
+
+func (c *TokenController) GetRedisStorage() *storage.RedisStorage {
+	return c.storage
+}
+
+// resources
+func (c *TokenController) forcedTokenGenerationHandler(w http.ResponseWriter, r *http.Request, params map[string]string) {
```
Comment (ArturKlajnerok) [inline controllers/token_controller.go]: can we just name it `tokenGenerationHandler` ? :smile_cat:

**Comment by ArturKlajnerok** [comment]
Please merge master so test should pass, other than that LGTM :+1:

**Pull request merged**